### PR TITLE
Adding 'python-libdiscid'

### PR DESCRIPTION
**Pull request by HeavyTony2**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by HeavyTony2** [comment]
@conda-forge/help-python-c Should I need to add a `build.sh` script? The build is asking for `Python.h`
https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=804480&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=230

**Comment by HeavyTony2** [comment]
Recap: Build won't work without cython in the build requirement and without the separate build script. However, the packaging step will fail. 

```
Packaging python-libdiscid
Packaging python-libdiscid-2.0.3-py311h459d7ec_0
WARNING: No files or script found for output python-libdiscid
Traceback (most recent call last):
  File "/home/conda/staged-recipes-copy/.ci_support/build_all.py", line 228, in <module>
    build_all(os.path.join(root_dir, "recipes"), args.arch)
  File "/home/conda/staged-recipes-copy/.ci_support/build_all.py", line 118, in build_all
    build_folders(recipes_dir, folders, arch, channel_urls)
  File "/home/conda/staged-recipes-copy/.ci_support/build_all.py", line 174, in build_folders
    conda_build.api.build([recipe], config=get_config(arch, channel_urls))
  File "/opt/conda/lib/python3.10/site-packages/conda_build/api.py", line 253, in build
    return build_tree(
  File "/opt/conda/lib/python3.10/site-packages/conda_build/build.py", line 3804, in build_tree
    packages_from_this = build(
  File "/opt/conda/lib/python3.10/site-packages/conda_build/build.py", line 2880, in build
    newly_built_packages = bundlers[pkg_type](output_d, m, env, stats)
  File "/opt/conda/lib/python3.10/site-packages/conda_build/build.py", line 1977, in bundle_conda
    raise CondaBuildException(
conda_build.exceptions.CondaBuildException: Empty package; python present in build and host deps.  You probably picked up the build environment's python  executable.  You need to alter your recipe to  use the PYTHON env var in your recipe to run that executable.
```

**Comment by HeavyTony2** [comment]
Other recap from test I made elsewhere
```
#!/bin/sh

set -ex

export CYLP_USE_CYTHON=1

python -m setup bdist_wheel -vvv
python -m pip install dist/*.whl -vvv
```
This would get me the same problem

However, if I tell the script to install the wheel file aready built, in other words ``python -m pip install alreadybuiltwheel.whl" it does work. So it's really an ENV setting issue.

**Comment by HeavyTony2** [comment]
About https://github.com/conda-forge/staged-recipes/pull/24262/commits/5ed439b507ed5482f308f499c10126c0ed7d076f
I suddenly realized the reason why the build python was working, but not the host python was surely related to the fact some dependancies were not listed in the build requirement.

**Comment by HeavyTony2** [comment]
@conda-forge/help-python-c ready for review

File: recipes/python-libdiscid/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  host:
+    - cython >=0.15
+    - python
+    - setuptools >=43
+  # - pkgconfig Is only an optional dependancy and happens to make the build fail
+    - pip
+    - libdiscid
+  run:
+    - python
+    - libdiscid
```
Comment (xhochy) [inline recipes/python-libdiscid/meta.yaml:29]: Should be removed as libdiscid has a run_exports.
```suggestion
```

**Review by xhochy — CHANGES_REQUESTED** [review]
One nit, otherwise lgtm.

**Review by xhochy — APPROVED** [review]

**Pull request merged**